owasp-top10-2017-apps/a1/copy-n-paste/README.md

```diff
@@ -1,40 +1,166 @@
 # CopyNPaste API
-> This is a simple Golang API that contains an example of an Injection vulnerability.
 
 <img src="images/CopyNPaste.png" align="center"/>
 
-## What is Injection?
+CopyNPaste is Golang web application that uses an API and a simple front end to simulate a login page. It has both `/register` and `/login` routes that, by communicating with a MySQL database, enables users register and enter into a generic system.
+
+## Index
 
-Definition from [OWASP](https://www.owasp.org/images/7/72/OWASP_Top_10-2017_%28en%29.pdf.pdf):
+- [Definition](#what-is-injection)
+- [Setup](#setup)
+- [Attack narrative](#attack-narrative)
+- [Objectives](#secure-this-app)
+- [Solutions](#pr-solutions)
+- [Contributing](#contributing)
+
+## What is Injection?
 
 Injection flaws, such as SQL, NoSQL, OS, and LDAP injection, occur when untrusted data is sent to an interpreter as part of a command or query. The attacker’s hostile data can trick the interpreter into executing unintended commands or accessing data without proper authorization.
 
-## Requirements
+The main goal of this project is to discuss how **SQL Injection** vulnerabilities can be exploited and to encourage developers send Pull Requests to secDevLabs on how they would mitigate these flaws.
+
+## Setup
+
+To start this intentionally **insecure application**, you will need [Docker][Docker Install] and [Docker Compose][Docker Compose Install]. After forking [secDevLabs](https://github.com/globocom/secDevLabs), you must type the following commands to start:
+
+```sh
+cd secDevLabs/owasp-top10-2017-apps/a1/copy-n-paste
+```
+
+```sh
+make install
+```
 
-To build this lab you will need [Docker][Docker Install] and [Docker Compose][Docker Compose Install].
+Then simply visit [localhost:3000][App], as exemplified below:
+
+<img src="images/CopyNPaste.png" align="center"/>
 
-## Deploy and Run
+## Get to know the app 💉
 
-After cloning this repository, you can type the following command to start the vulnerable application:
+To properly understand how this application works, you can follow these simple steps:
 
+- Register a new user via front-end.
+- Login as this user via front-end.
+* Register another user now using command line:
 ```sh
-$ make install
+curl -s -H "Content-Type: application/json" -d '{"user":"bob", "pass":"password", "passcheck":"password"}' http://localhost:3000/register
 ```
+* Login as this second user now using command line:
+```sh
+curl -s -H "Content-Type: application/json" -d '{"user":"bob", "pass":"password"}' http://localhost:3000/login
+
+```
+
+## Attack narrative
+
+Now that you know the purpose of this app, what could possibly go wrong? The following section describes how an attacker could identify and eventually find sensitive information about the app or it's users. We encourage you to follow these steps and try to reproduce them on your own to better understand the attack vector! 😜
+
+### 👀
+
+#### Lack of input validation allows for remote code execution
```

Comment:
I think this lack of input validation allows _SQL Injection attacks_. You may eventually (or not) find a way to escalate this vulnerability into a RCE but, by taking a look at what evidences we have, `SQL Injection attacks` would be a better description to this attack.

Reply:
Done in 43ce9ce